x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -1405,40 +1548,73 @@ public boolean maybeEvictLeastUsed() {
         }
 
         private void computeDecay() {
+            long now = System.currentTimeMillis();
+            long afterLock;
+            long end;
             synchronized (SharedBlobCacheService.this) {
-                long now = relativeTimeInMillis();
-                for (int i = 0; i < maxFreq; i++) {
-                    for (LFUCacheEntry entry = freqs[i]; entry != null; entry = entry.next) {
-                        if (entry.freq > 0 && now - entry.lastAccessed >= 2 * minTimeDelta) {
-                            unlink(entry);
-                            entry.freq--;
-                            pushEntryToBack(entry);
-                        }
-                    }
+                afterLock = System.currentTimeMillis();
+                appendLevel1ToLevel0();
+                for (int i = 2; i < maxFreq; i++) {
+                    assert freqs[i - 1] == null;
+                    freqs[i - 1] = freqs[i];
+                    freqs[i] = null;
+                    decrementFreqList(freqs[i - 1]);
+                    assert freqs[i - 1] == null || invariant(freqs[i - 1], true);
                 }
             }
+            end = System.currentTimeMillis();
+            logger.info("Decay took {} ms (acquire lock: {} ms)", end - now, afterLock - now);
```

Comment:
Just a reminder to lower this to debug, and maybe use `threadPool.relativeTimeInMillis()` instead of `System.currentTimeMillis()`

Reply:
I changed it to `threadPool.rawRelativeTimeInMillis()`, since the other one has a higher granularity.